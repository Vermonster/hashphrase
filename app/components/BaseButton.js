import React from 'react';
import { Button, Text } from 'react-native-paper';
import PropTypes from 'prop-types';

const BaseButton = ({ onPress, buttonText }) => (
  <Button mode="contained" onPress={onPress}>
    <Text>{ buttonText }</Text>
  </Button>
);

BaseButton.propTypes = {
  onPress: PropTypes.func.isRequired,
};

export default BaseButton;
