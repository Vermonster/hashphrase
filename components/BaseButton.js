import React from 'react';
import { Button, Text } from 'native-base';
import PropTypes from 'prop-types';

const BaseButton = ({onPress, buttonText}) => (
  <Button full onPress={onPress}>
    <Text>{buttonText}</Text>
  </Button>
);

BaseButton.propTypes = {
  onPress: PropTypes.func.isRequired,
}

export default BaseButton;