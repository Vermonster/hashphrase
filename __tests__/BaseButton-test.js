import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import { shallow, configure } from 'enzyme';
import { Button, Text } from 'native-base';
import BaseButton from '../components/BaseButton';

configure({ adapter: new Adapter() });

const mock = jest.fn();
const innerText = 'Generate Password';

describe('<BaseButton />', () => {
  it('should deep render without crashing', () => {
    const tree = renderer.create(<BaseButton onPress={mock} buttonText={innerText} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should shallow render without crashing', () => {
    const wrapper = shallow(<BaseButton onPress={mock} buttonText={innerText} />);
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it('should render the correct components', () => {
    const wrapper = shallow(<BaseButton onPress={mock} buttonText={innerText} />);
    expect(wrapper.find(Button)).toHaveLength(1);
    expect(wrapper.find(Text)).toHaveLength(1);
  });

  it('should execute onPress function', () => {
    const onPressEvent = jest.fn();
    onPressEvent.mockReturnValue('onPress invoked');
    const wrapper = shallow(<BaseButton onPress={onPressEvent} buttonText={innerText} />);
    wrapper.find(Button).simulate('press');
    expect(onPressEvent.mock.calls.length).toBe(1);
  });
});
