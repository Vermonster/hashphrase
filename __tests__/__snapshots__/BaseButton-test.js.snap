// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BaseButton /> should deep render without crashing 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "stretch",
      "backgroundColor": "#007aff",
      "borderBottomWidth": null,
      "borderColor": "#007aff",
      "borderLeftWidth": null,
      "borderRadius": 0,
      "borderRightWidth": null,
      "borderTopWidth": null,
      "elevation": 2,
      "flexDirection": "row",
      "height": 45,
      "justifyContent": "center",
      "opacity": 1,
      "paddingBottom": 6,
      "paddingTop": 6,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "backgroundColor": "transparent",
        "color": "#fff",
        "fontFamily": "System",
        "fontSize": 16.5,
        "marginLeft": 0,
        "marginRight": 0,
        "paddingLeft": 16,
        "paddingRight": 16,
      }
    }
    uppercase={false}
    virtual={undefined}
  >
    Generate Password
  </Text>
</View>
`;

exports[`<BaseButton /> should shallow render without crashing 1`] = `
"<Styled(Button) full={true} onPress={[Function: mockConstructor]} virtual={[undefined]}>
  <Styled(Text) virtual={[undefined]}>
    Generate Password
  </Styled(Text)>
</Styled(Button)>"
`;
